src/bootstrap_cache.rs

```diff
@@ -0,0 +1,163 @@
+use crate::dirs::Dirs;
+use crate::{Error, NodeInfo, R};
+use bincode::{deserialize_from, serialize_into};
+use log::{error, info};
+use std::fs::{self, File};
+use std::io::{self, BufReader, BufWriter};
+use std::path::Path;
+use std::path::PathBuf;
+
+/// Maximum peers in the cache.
+const MAX_CACHE_SIZE: usize = 200;
+
+type Stack = Vec<NodeInfo>;
+
+/// A very simple LRU like struct that writes itself to disk every 10 entries added.
+pub struct BootstrapCache {
```

Comment:
We should perhaps pass the hard-coded-contacts list to this to be held as a member-var. That way it can always see if the peer being asked to add belongs to HCC. If yes then ignore.